MennaMohie/assignment4/Solution.cpp

```diff
@@ -0,0 +1,115 @@
+#include<iostream>
+#include<vector>
+#include"Testing.h"
+using namespace std;
+
+class Counter
+{
+	int count, rows, cols;
+	vector<vector<bool>> grid;
+	int deltaX[4] = { -1,0,0,1 };
+	int deltaY[4] = { 0,-1,1,0 };
+
+	bool validRow(const int &row)
```

Comment:
Some comment on style since I don't have much else :)
It may be more legible to put the '{' in the previous line, so more stuff fits on the screen at once.